task/router.go

```diff
@@ -99,6 +100,12 @@ func (h *Router) Handle(ctx context.Context, req *Request) Response {
 	// add the structured logger to the context.
 	ctx = logger.WithContext(ctx, log)
 
+	// Set the task logger to stdout if it is not set.
+	// A custom logger can be set by adding a middleware to the router.
+	if req.Logger == nil {
+		req.Logger = os.Stdout
```

Comment:
should this be set to `os.Stdout` or `io.Discard` by default?  I feel like libraries should not write to stderr / stdout by default.  It should be something the end-user has to configure.

Reply:
alternatively we could do something like this:

```diff
type Router struct {
	middleware []func(Handler) Handler
	handlers   map[string]Handler
	notfound   Handler
+	logger io.Writer 
}

func NewRouter() *Router {
	return &Router{
		handlers: map[string]Handler{},
+		logger: io.Discard,
	}
}


+func (r *router) Logger(w io.Writer) {
+	r.logger = w
+}
```

and then

```Go
// use default logger if logger is nil
if req.Logger == nil {
		req.Logger = r.logger
```

Reply:
Makes sense @bradrydzewski  - changed it to `io.Discard` instead.
Have skipped adding a logger at the router level since we can handle that using logging middleware.